### PR TITLE
Consistent job naming

### PR DESCRIPTION
**Pull request by bastjan**
## Summary

This PR:
- Prefixes the job with the backup type 
- Sets the type as the container name

Part of #427.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [x] Update the documentation. (Looked through the docs, the generated batch/v1.Job name is referenced nowhere)
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: job/job.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func GenerateGenericJob(obj k8upv1alpha1.JobObject, config Config) (*batchv1.Job
 					RestartPolicy: corev1.RestartPolicyOnFailure,
 					Containers: []corev1.Container{
 						{
-							Name:      obj.GetMetaObject().GetName(),
+							Name:      string(obj.GetType()),
```
Comment (ccremer) [inline job/job.go:68]: Is there no String() method available?
```suggestion
							Name:      obj.GetType().String(),
```

File: api/v1alpha1/schedule_types.go [hunk 2]
```diff
@@ -119,6 +119,11 @@ func (s *Schedule) GetMetaObject() metav1.Object {
 	return s
 }
 
+// GetJobName returns the name of the underlying batch/v1 job.
+func (s *Schedule) GetJobName() string {
+	return s.GetType().String() + "-" + s.Name
+}
```
Comment (ccremer) [inline api/v1alpha1/schedule_types.go:125]: Godoc comment doesn't match here. There's no batch/v1 job

File: api/v1alpha1/job_name_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		&v1alpha1.Archive{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Backup{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Check{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Prune{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Restore{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		&v1alpha1.Schedule{ObjectMeta: metav1.ObjectMeta{Name: name}},
+	}
+
+	for _, subject := range subjects {
+		assert.Equal(t, subject.GetType().String()+"-"+name, subject.GetJobName())
```
Comment (corvus-ch) [inline api/v1alpha1/job_name_test.go:23]: I would prefer to not compute the expected value but have it hard coded. For that I would make `subjects` a dictionary containing both the expected job name and the subject (`map[string]v1alpha1.JobObject`). While at it, I also would use nested test functions. This makes it esier to identify the exact case when there is a test failure.

**Review by corvus-ch — APPROVED** [review]
Whith the comments by @ccremer addressed, this is LGTM.

File: job/job.go (hunk 1 above)
Comment (bastjan) [reply job/job.go:68]: there is one 👍 looks like my editor thought otherwise, thanks 😊

**Review by ccremer — APPROVED** [review]

**Pull request merged**